addons/osio-addon/src/main/java/io/fabric8/launcher/osio/steps/OpenShiftSteps.java

```diff
@@ -97,23 +100,33 @@ public void ensureCDGithubSecretExists() {
     }
 
     public BuildConfig createBuildConfig(OsioProjectile projectile, GitRepository repository) {
-        BuildConfig buildConfig = createBuildConfigObject(projectile, repository);
-        String spaceName = projectile.getSpace().getName();
-        String gitOrganizationName = projectile.getGitOrganization();
-        setSpaceNameLabelOnPipeline(spaceName, buildConfig);
-
-        // set the value reponame if repo is in user's namespace otherwise set orgname.reponame
-        if (gitOrganizationName != null) {
-            setGitRepositoryNameLabelOnPipeline(gitOrganizationName + "." + projectile.getGitRepositoryName(), buildConfig);
-        } else {
-            setGitRepositoryNameLabelOnPipeline(projectile.getGitRepositoryName(), buildConfig);
-        }
+        final String spaceName = projectile.getSpace().getName();
+        final String gitOrganizationName = projectile.getGitOrganization();
+        BuildConfig buildConfig = new BuildConfigBuilder(createBuildConfigObject(projectile, repository))
+                .accept(new TypedVisitor<ObjectMetaBuilder>() {
+                    @Override
+                    public void visit(ObjectMetaBuilder o) {
+                       o.addToLabels("space", spaceName);
+                       o.addToLabels("openshift.io/gitRepository",
+                               gitOrganizationName != null
+                                       ? gitOrganizationName + projectile.getGitRepositoryName()
```

Comment:
I think it's missing a dot `.` between the gitOrganizationName and the Git repository name?